pandas/core/groupby/grouper.py

```diff
@@ -527,13 +520,10 @@ def __init__(
                 self.grouping_vector = Index(ng, name=newgrouper.result_index.name)
 
         elif is_categorical_dtype(self.grouping_vector):
-            # a passed Categorical
-            self._passed_categorical = True
-
-            self.grouping_vector, self._all_grouper = recode_for_groupby(
+            self._cat_grouper = CategoricalGrouper.make(
                 self.grouping_vector, sort, observed
             )
-
+            self.grouping_vector = self._cat_grouper.new_grouping_vector
```

Comment:
See comment above, could change this to just `self._cat_grouper, self.grouping_vector = CategoricalGrouper.make(...)`